### PR TITLE
bakery: concurrency-safe ThirdPartyLocator

### PR DESCRIPTION
**Pull request by rogpeppe**
In general, ThirdPartyLocator implementations must be
concurrency safe, so document them as such and
fix bakery.ThirdPartyStore to work OK when its
methods are called concurrently.

### CODE REVIEW

**Review by mhilton — APPROVED** [review]
LGTM

File: bakery/keys.go [hunk 1]
```diff
@@ -98,13 +99,15 @@ type ThirdPartyInfo struct {
...
 	// party at the given location. It returns ErrNotFound if no match is found.
+	// This method must be safe to call concurrently.
 	ThirdPartyInfo(ctx context.Context, loc string) (ThirdPartyInfo, error)
 }
 
 // ThirdPartyStore implements a simple ThirdPartyLocator.
 // A trailing slash on locations is ignored.
 type ThirdPartyStore struct {
-	m map[string]ThirdPartyInfo
+	mu sync.Mutex
```
Comment (alesstimec) [inline bakery/keys.go]: you could use a RWMutex

File: bakery/keys.go [hunk 2]
```diff
@@ -127,6 +133,8 @@ func canonicalLocation(loc string) string {
 
 // ThirdPartyInfo implements the ThirdPartyLocator interface.
 func (s *ThirdPartyStore) ThirdPartyInfo(ctx context.Context, loc string) (ThirdPartyInfo, error) {
+	s.mu.Lock()
```
Comment (alesstimec) [inline bakery/keys.go]: and do a RLock, RUnlock here

**Review by alesstimec — APPROVED** [review]
LGTM with a comment

File: bakery/keys.go (hunk 1 above)
Comment (rogpeppe) [reply bakery/keys.go]: I usually don't consider RWMutexes unless the mutex is likely to be highly contended (e.g. the read or the write path is slow), but why not? Done.

**Pull request merged**